reference/ast-examples.md

```diff
@@ -0,0 +1,185 @@
+
+On the other hand, you _could_ define a [Binary](#binary) with other Binaries inside of that. That would be a valid AST but it **would not** represent valid Java code as just entering `1+2` (a valid Binary) in a Java file would not compile (that code would need to be in an expression which is then in a class to be valid).
+
+> Every object in the [code section](#code) is an AST in the sense that it implements a Tree interface. These objects, however, need to be combined in an appropriate way to represent valid Java. It is your duty as a recipe author to ensure that the final AST represents valid Java code.
+
+## Code
+
+Below is a simple Java class whose entire purpose is to demonstrate different types of ASTs. Each of the following sections will highlight different parts of this code to demonstrate which chunks correspond to which AST.
+
+```java
+package org.openrewrite;
+                
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@AnAnnotation
+public class A {
+
+    List<Integer> a = new ArrayList<>();
+
+    int foo() {
+        int a = 1 + 2, b = 3, c;
+        this.a = this.a.stream()
+                .map(it -> it + 1)
+                .collect(Collectors.toList());
+        return a;
+    }
+}
+
+@interface AnAnnotation {}
+```
+
+### Binary
+
+A [Binary AST](https://github.com/openrewrite/rewrite/blob/main/rewrite-java/src/main/java/org/openrewrite/java/tree/J.java#L597-L705) is an [Expression](#expression) with a left and right side, separated by an operator. Examples of operators include `+`, `-`, `||`, `&&`, and more.
+
+![Binary Example](<../.gitbook/assets/Binary.png>)
+
+### Block 
+
+A [Block AST](https://github.com/openrewrite/rewrite/blob/main/rewrite-java/src/main/java/org/openrewrite/java/tree/J.java#L712-L851) is a pair of curly braces and the [Statements](#statement) contained within. Blocks can be nested inside of each other.
+
+![Block Example](<../.gitbook/assets/Block.png>)
+
+### ClassDeclaration
+
+A [ClassDeclaration AST](https://github.com/openrewrite/rewrite/blob/main/rewrite-java/src/main/java/org/openrewrite/java/tree/J.java#L1062-L1336) contains all of the code for any Java class. Please note that a `ClassDeclaration` can be nested inside of another class such as with:
+
+```java
+public class A {
+    // ...
+    private class B {
+        // ...
+    }
+}
+```
+
+![ClassDeclaration Example](<../.gitbook/assets/ClassDeclaration.png>)
+
+### CompilationUnit
+
+A [CompilationUnit AST](https://github.com/openrewrite/rewrite/blob/main/rewrite-java/src/main/java/org/openrewrite/java/tree/J.java#L1342-L1545) is the root of the Java AST. In order for an AST to represent valid Java code, all other elements must be contained inside of this. 
```

Comment:
I don't think we need to say "<AST type> AST" for each of these. It is redundant. 

It's like if we were listing breeds of dogs like this:

- Labrador dog
- Schnauzer dog
- Beagle dog
- Dachshund dog
- Corgi dog

Probably didn't need so many dog types to illustrate my point. But look how cute this beagle is:
![image](https://user-images.githubusercontent.com/8549033/206067786-7add3347-569c-4ba3-b1e3-742ed3a74826.png)

Reply:
I'm okay with making this change -- but I do want to call out that I'm not sure I agree with that. The naming conventions we have don't make it obvious that these things _are_ ASTs. If I saw a section called, `Identifier` and saw that it talked about variable names, I definitely would not assume it's an AST. That being said, I recognize that I'm still learning this so will default to you on this.